app/chapters/[slug]/layout.tsx

```diff
@@ -1,13 +1,37 @@
-import Navbar from 'components/Navbar'
+'use client'
+
+import Navbar, { NavbarMobile } from 'components/Navbar'
 import chapters from 'content/chapters'
+import { useEffect, useState } from 'react'
+
+function getWindowWidth() {
+  const { innerWidth: width } = window
+  return width
+}
 
-export default async function Layout({ children, params }) {
+export default function Layout({ children, params }) {
   const isValidChapter = chapters[params.slug]
 
+  const desktopSize = 800
+
+  const windowWidth = getWindowWidth()
+  const [isDesktopSize, setIsDesktopSize] = useState(windowWidth > desktopSize)
```

Comment:
Let's just use `window.innerWidth` directly here instead of creating a function for it (`getWindowWidth`).

Reply:
Also now that I think about it, please use the `useMediaQuery` hook, you can find an example in `TerminalChallenge.tsx`